### ISSUE TITLE
app will randomly come from minimized to dock on it's own

### ISSUE THREAD

**Issue by recoi1er** [SchizoDuckie/DuckieTV#1208]
**What build of DuckieTV are you using (Standlone / Chrome Extension (New Tab / Browser Action))**
Standalone

**What version of DuckieTV are you using (Stable 1.1.x / Nightly yyyymmddHHMM)**
Nightly v201908240130

**What is your Operating System (Windows, Mac, Linux, Android)**
MacOS

**Describe the problem you are having and steps to reproduce if available**
Computer has screen on, either being used or not, DuckieTV is minimized. DuckieTV will come out of the dock on it's own. No alert, no interaction required. Just randomly appears

**Attach any DuckieTV statistics or Developer Console logs if available**
...

**Comment by garfield69** [comment]
are you running your mac 24/7 and have Dtv active all the time?

**Comment by garfield69** [comment]
There is an Internal Daily Reset that DuckieTV performs a couple seconds after midnight so that all the schedules and calendar events are set for the new day.       

If you want to avoid DuckieTV grabbing focus during this event, then try the following: 
Navigate to the *Settings > Window* side panel, and set the *Window on Start-up* to **Minimised** by clicking that button.     
You may also want to tick the *System Tray* check-boxes for *Always show Tray Icon* and *Minimise Button Minimises DuckieTV to Tray*.      
![image](https://user-images.githubusercontent.com/6933240/33219294-5824e230-d1a6-11e7-8de9-1fc775f1decf.png)

if you still get this issue after these setting changes, then you may also need to set the notifications off.
![image](https://user-images.githubusercontent.com/6933240/65813194-d2cf9e80-e225-11e9-884c-2c5285eb573c.png)


let me know if you still get this problem after that.

[edit] oops, pasted wrong image.

**Comment by recoi1er** [comment]
My Mac is typically always on, but I reboot regularly (even had rebooted yesterday before this started occurring as well having had created a brand new profile 2 days agi for another issue I am having with Mac Mail app). Dtv typically is always open, I have been using it this way for about a year now

**Comment by garfield69** [comment]
just check/set the settings as I described earlier and see if that helps.

**Comment by recoi1er** [comment]
Oh I didn’t think that applied since it happened before midnight, I’ll try that out though (already set to minimize in system prefs but that feature is one of the worst that Apple has made - just don’t work)

Turning off notifications entirely is not something I want to do, is that really a possible solution? Lose full notification functionality over a bug that isn’t messing with notifications?

**Comment by garfield69** [comment]
I'm only providing you with options to try out. 
do the first set and see if it makes a difference. 
if it does not make a difference, then try switching off the notifications.
see if that make a difference.
if it does not, then turn it back on. 
We have not made any changes to the Dtv AngularJS code that impacts on windows and notifications.
We did bump the Standalone NWJS engine from NWJS 0.39.3(with Chromium 75) to NWJS 0.40.2(with Chromium 76) on the 1st of September.
If there has been any changes to the way NWJS works, then that is out of our control.
As you may know Dtv is written in javascript using the AngularJS framework, as a Chrome Browser Extension. The standalone version is the same Dtv code running under the NWJS engine which uses an internal chromium browser to allow Dtv to work standalone.

**Comment by recoi1er** [comment]
Gotcha, thank you for the info!

I checked and the minimize setting was enabled but I unchecking and rechecked it. I’ll keep note if it happens again!

**Comment by recoi1er** [comment]
Hello!

I just opened your app and it’s blank! I restored a back up and no change! I also reinstalled the newest nightly and also no change. Rebooted. Etc. Even going back to other months, it’s all blank!

Any ideas?


Thank you,

Oliver Shipp

**Comment by garfield69** [comment]
Urg. I have see this once before on Mac. 
Some cleanup tool (system, anti-malware, security update, I don't know what) has blown away something in the data-space that Dtv is using, and Dtv no longer has the profile it needs to access the DB :-(
Assuming you have a Dtv Backup of the settings and series taken with the Dtv > Settings > Backup previously then best recovery is to
- shutdown Dtv
- delete the data folder `~/Library/Application Support/DuckieTV-Standalone`
- start dtv and wait a minute or two for the pre-init to complete and settle down
- at the `you have no series yet` page import your backup.
- once the backup is complete, shutdown and restart Dtv to be sure all the settings are set.

**Comment by recoi1er** [comment]
Sweet that did it! Thank you!

I did just use an app called CampTune to resize my Mac and Bootcamp drives and thats when it happened?

Going forward since I have a time machine backup, do you know what file/folder I could restore to bring that back without having to rebuild and lose changes from the last backup to now?


Thank you,

Oliver Shipp

**Comment by garfield69** [comment]
Basically the whole data folder. there are far too many variables to identify specifically.

**Comment by recoi1er** [comment]
Gotcha ok, if it ever happens again I will try that out!
Thanks


Thank you,

Oliver Shipp
